flask_multipass/providers/ldap/util.py

```diff
@@ -135,7 +135,7 @@ def ldap_connect(settings, use_cache=True):
     ldap_connection = ReconnectLDAPObject(settings['uri'], bytes_mode=False)
     ldap_connection.protocol_version = ldap.VERSION3
     ldap_connection.set_option(ldap.OPT_REFERRALS, 0)
-    ldap_connection.set_option(ldap.OPT_X_TLS, ldap.OPT_X_TLS_DEMAND if use_ldaps else ldap.OPT_X_TLS_NEVER)
+    ldap_connection.set_option(ldap.OPT_X_TLS_DEMAND if use_ldaps else ldap.OPT_X_TLS_NEVER, True)
```

Comment:
Just wondering, where is this somewhat strange new usage documented?

According to the documentation, I'd expect setting `ldap.OPT_X_TLS_REQUIRE_CERT` to either `OPT_X_TLS_NEVER` or `OPT_X_TLS_DEMAND`...

Reply:
Actually, we already have this:

```
    ldap_connection.set_option(ldap.OPT_X_TLS_REQUIRE_CERT,
                               ldap.OPT_X_TLS_DEMAND if settings['verify_cert'] else ldap.OPT_X_TLS_ALLOW)
```

So the suggestion from my last comment is already there.

https://www.python-ldap.org/en/python-ldap-3.4.2/reference/ldap.html#ldap.initialize

> The internal connection type is determined from the URI, TCP for ldap:// / ldaps://, [...]

But I have no idea if this just indicates the socket type or also whether to use SSL/TLS from the beginning (instead of the nowadays much more common STARTTLS)...

Reply:
So looking at the openldap code, I think using an `ldaps://` url may be all that's needed... https://github.com/openldap/openldap/blob/98a0029daeb8aaa7bc58428ad3f94eface7f997b/libraries/libldap/open.c#L650

Reply:
Thanks! I will look at this further and return to this!

Reply:
You are completely right, I have amended the merge request.